notebooks/Rules.py

```
@@ -0,0 +1,93 @@
+        self.df['Date_Time'] = pd.to_datetime(self.df['Date_Time'], format='%Y-%m-%d %H:%M:%S')
+
+    def rule1_n_transactions_m_minutes(self, max_transactions=3, max_minutes=30):
+        flagged = []
+        for _, row in self.df.iterrows():
+            dt = row['Date_Time']
+            recent = self.df[(self.df['Date_Time'] <= dt) & (self.df['Date_Time'] >= dt - pd.Timedelta(minutes=max_minutes))]
+            if len(recent) > max_transactions:
+                flagged.append(row['Transaction_Id'])
+        return flagged
+
+    def rule2_high_risk_cities(self, high_risk=['Sindh', 'Balochistan']):
```

Ruff (0.8.2)

Do not use mutable data structures for argument defaults

Replace with `None`; initialize within function

(B006)